lib/common/s3.sh

```diff
@@ -68,12 +68,32 @@ s3_put_no_index_keep_file() {
     local src=$1; shift
     local object_name=$1; shift
     local dst=$S3_BUCKET/$object_name
+    local mime=''
 
     test -f $src || file_error "Not a regular file"
 
+    if [[ $src =~ \.gz$ ]]; then
+        mime='application/gzip'
+    elif [[ $src =~ \.nc$ ]]; then
+        mime='application/octet-stream'
+    elif [[ $src =~ \.csv$ ]]; then
+        mime='text/csv'
+    elif [[ $src =~ \.pdf$ ]]; then
+        mime='application/pdf'
+    elif [[ $src =~ \.png$ ]]; then
+        mime='image/png'
+    elif [[ $src =~ \.jpg$ ]]; then
+        mime='image/jpeg'
+    fi
```

Comment:
@lbesnard @pblain 

> Maybe we could have a special octet-stream for any other extension.

The default is already octet-stream for any other extension.

> and having to reprocess all files because of this will take some time


I have java code that can update metadata recursively across all s3 keys. 

If this had been merged as I suggested then I could have updated everything at the end of last iteration when I had nothing else to do.

Issue is closed now, so it's your problem.